api/app/models/spree/api/webhooks/order_decorator.rb

```diff
@@ -15,6 +19,17 @@ def finalize!
         def after_resume
           super
           queue_webhooks_requests!('order.resumed')
+          self.state_machine_resumed = false
```

Comment:
@sebastian-palma why do we need here? shouldn't this be in the Order class?

Reply:
That has a example case [here](https://github.com/spree/spree/blob/182ee489a79231706404a55b489b710197948a44/api/spec/models/spree/api/webhooks/order_decorator_spec.rb#L45:L52). As far as I saw, when the state machine event `resume` is executed two updates occur in the background, one to update the receiver `state` column value, and the other to create the state change record, and both operations count as an update, triggering almost at the same time and with a slight difference in the payload/body of the webhook. So this is just a flag to know whether the resume event was executed and avoid sending that webhook twice @damianlegawiec.

> shouldn't this be in the Order class?

It was added as a virtual attribute, just for this case.

Reply:
Got it, thanks for the explanation 👍